### PR TITLE
feat: Profile page integrate API

### PR DESCRIPTION
**Pull request by whDongRui**

### CODE REVIEW

File: src/pages/Admin/Users/index.jsx [hunk 1]
```diff
@@ -240,7 +240,7 @@ export default class Users extends Component {
         onCancel={userStore.hideModal}
         hideFooter
       >
-        <form className="formContent" onSubmit={userStore.createOrModify} method="post">
+        <form className="formContent" onSubmit={e => userStore.createOrModify(e)} method="post">
```
Comment (iwisunny) [inline src/pages/Admin/Users/index.jsx]: no need to change

File: server/routes/api.js [hunk 2]
```diff
@@ -39,16 +39,13 @@ router.post('/api/*', async ctx => {
...
-        .post([apiServer, 'oauth2/token'].join('/'))
-        .set(header)
-        .send({
-          grant_type: 'refresh_token',
-          client_id: ctx.store.clientId,
-          client_secret: ctx.store.clientSecret,
-          scope: '',
-          refresh_token: refresh_token
-        });
+      const res = await agent.post([apiServer, 'oauth2/token'].join('/')).send({
```
Comment (iwisunny) [inline server/routes/api.js]: ```agent.post(url, {...})```

File: src/pages/Profile/index.jsx [hunk 3]
```diff
@@ -14,17 +15,19 @@ import styles from './index.scss';
...
-  }
-
   constructor(props) {
     super(props);
     this.state = {
       currentForm: 'basic'
     };
   }
 
+  componentDidMount() {
```
Comment (iwisunny) [inline src/pages/Profile/index.jsx]: put this logic in `onEnter`, or SSR will not work

File: src/pages/Profile/index.jsx [hunk 4]
```diff
@@ -35,64 +38,97 @@ export default class Profile extends Component {
 
   renderBasic() {
     const { userStore, t } = this.props;
-    const { userDetail, changeUser, changeUserRole } = userStore;
+    const { userDetail, changeUser, modifyUser } = userStore;
+    const emailRegexp =
+      "[\\w!#$%&'*+/=?^_`{|}~-]+(?:\\.[\\w!#$%&'*+/=?^_`{|}~-]+)*@(?:[\\w](?:[\\w-]*[\\w])?\\.)[\\w](?:[\\w-]*[\\w])?";
```
Comment (iwisunny) [inline src/pages/Profile/index.jsx]: regExp is too complex ?

File: src/pages/Profile/index.jsx [hunk 5]
```diff
@@ -35,64 +38,97 @@ export default class Profile extends Component {
...
 
     return (
-      <div className={styles.form}>
+      <form className={styles.form} onSubmit={e => modifyUser(e)} method="post">
```
Comment (iwisunny) [inline src/pages/Profile/index.jsx]: no need to pass event to onSubmit

File: src/pages/Admin/Users/index.jsx (hunk 1 above)
Comment (whDongRui) [reply src/pages/Admin/Users/index.jsx]: This change for prevent page refresh。
 createOrModify = async e => {
    e.preventDefault();
    ...
}

File: src/pages/Profile/index.jsx (hunk 3 above)
Comment (whDongRui) [reply src/pages/Profile/index.jsx]: If put in onEnter ，refresh page getCookie method will error .

File: src/pages/Profile/index.jsx (hunk 5 above)
Comment (whDongRui) [reply src/pages/Profile/index.jsx]: This change for prevent page refresh when submit。

File: server/routes/api.js (hunk 2 above)
Comment (whDongRui) [reply server/routes/api.js]: ok, define url variable

File: src/pages/Profile/index.jsx (hunk 4 above)
Comment (whDongRui) [reply src/pages/Profile/index.jsx]: ok.

**Comment by iwisunny** [comment]
fix: #348

**Pull request merged**